### ISSUE TITLE
[Bug] XAML compiler broke with 4.7.0

### ISSUE THREAD

**Issue by Dids** [xamarin/Xamarin.Forms#11101]
### Description

Updating to Xamarin.Forms 4.7.0 broke something in the XAML compilation, where every XAML file change fails to save.

Downgrading to the latest 4.6.x release fixes it, while restarting the IDE and/or deleting bin/obj directories does not.

Additionally, the application still compiles and runs fine without any errors or warnings, but any changes made to the XAML do not persist to the application itself.

Example errors from "Tool Output":
```
App.xaml : error :  : XamlC error XFC0000 : Cannot resolve type "Application".
AppShell.xaml : error :  : XamlC error XFC0000 : Cannot resolve type "Shell".
Views/AboutPage.xaml : error :  : XamlC error XFC0000 : Cannot resolve type "ContentPage".
Views/ItemDetailPage.xaml : error :  : XamlC error XFC0000 : Cannot resolve type "ContentPage".
Views/ItemsPage.xaml : error :  : XamlC error XFC0000 : Cannot resolve type "ContentPage".
Views/NewItemPage.xaml : error :  : XamlC error XFC0000 : Cannot resolve type "ContentPage".
```

### Steps to Reproduce

1. Create a new Xamarin.Forms Shell application
2. Update the dependencies (specifically Xamarin.Forms to 4.7.0)
3. Edit `AppShell.xaml` in any way and hit Save

### Expected Behavior

XAML should successfully save after any edits.

### Actual Behavior

XAML fails to generate files when saving changes.

### Basic Information

- Version with issue: 4.7.0
- Last known good version: 4.6.0.967
- IDE: Visual Studio for Mac (latest stable branch versions of everything)
- Platform Target Frameworks:
  - iOS:  13.5
- Nuget Packages: Xamarin.Forms. Xamarin.Essentials

### Reproduction Link

https://github.com/Dids/XAMLCompilationBug

### Workaround

Downgrade to Xamarin.Forms 4.6.0.967 or older.

**Comment by jsuarezruiz** [comment]
@StephaneDelcroix Could be related with https://github.com/xamarin/Xamarin.Forms/pull/11021?

**Comment by eddie010** [comment]
I'm currently running into this as well... Every page now generates the error above... I'm downgrading in the meantime :)

**Comment by StephaneDelcroix** [comment]
it is definitely related to the changes done on #11042 

Those errors happen on "UpdateDesignTimeXaml" the first time the IDE tries to update the files. But it doesn't affect the compilation of the project at all.

@Dids by
>Additionally, the application still compiles and runs fine without any errors or warnings, but any changes made to the XAML do not persist to the application itself.

you mean HotReload ?

**Comment by kaniosm** [comment]
@StephaneDelcroix, it doesn't affect the compilation but it does affect the functionality...
In my case any code shell uri's registered in code (i.e. Routing.RegisterRoute("SchedulerSettings", typeof(SchedulerSettingsPage)) will not work (routing doesn't work).

This is frustrating guys...
I removed a proxy project that was offering access to IMultibing since Multibinding is now part of 4.7, simplified all my grid rowdefinitions and columndefinitions and now I have to revert back to and switch to 4.6!

**Comment by StephaneDelcroix** [comment]
@kaniosm I'm not trying to produce bad excuses by saying it still builds. This issue, like any regression, is top priority for us. We're on this, we hope to get a fix really soon, and that'll be followed by a service release for 4.7.0.

can you explain a bit more about Registered routes no longer working ?

**Comment by kaniosm** [comment]
Hi @StephaneDelcroix, I’m not implying you are giving bad excuses. Sorry if it seemed so! I was just trying to clarify that even if it builds there are still issues that might be related to this. 
I have a few routes that are registered in code during startup and some routes defined in xaml within the shell page. All routes registered in code do not work anymore. No error is produced during runtime, but the hi will not redirect to the route.

**Comment by StephaneDelcroix** [comment]
@kaniosm that shell routes looks something different. could you please open a new issue, ideally with a small repro for it ? 

/cc @PureWeen

**Comment by jwoeste** [comment]
Hi - just to follow up:

Sample errors that arise from this issue:
`/Users/jenswoeste/Projects/lukka/curo/FrontEnd/Curo/Pages/StockPage.xaml: Error: The 'MSBuild:UpdateDesignTimeXaml' code generator crashed: A task was canceled.
  at MonoDevelop.Ide.CustomTools.MSBuildCustomTool.Generate (MonoDevelop.Core.ProgressMonitor monitor, MonoDevelop.Projects.Project project, MonoDevelop.Projects.ProjectFile file, MonoDevelop.Ide.CustomTools.SingleFileCustomToolResult result) [0x00046] in /Users/builder/azdo/_work/1/s/main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.CustomTools/MSBuildCustomTool.cs:53 
  at MonoDevelop.Ide.CustomTools.CustomToolService.UpdateAsync (MonoDevelop.Projects.ProjectFile file, MonoDevelop.Projects.Project project, System.Boolean force) [0x002ba] in /Users/builder/azdo/_work/1/s/main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.CustomTools/CustomToolService.cs:376 
/Users/jenswoeste/Projects/lukka/curo/FrontEnd/Curo/Pages/PortfolioDetailPage.xaml: Error:  : XamlC error XFC0000 : Cannot resolve type "Application". (Curo)
/Users/jenswoeste/Projects/lukka/curo/FrontEnd/Curo/Pages/PortfolioDetailPage.xaml: Error:  : XamlC error XFC0000 : Cannot resolve type "Shell". (Curo)`

It builds but MSBuild:UpdateDesignTimeXaml code generator crashes. This initially only affects hot reloads (and annoys my OCD as errors are show plenty). You can still pull off a bit of debugging. After a while the IDE gets quite flakey and triggers random freezes which is probably more an issue for the VS Mac team than the Xamarin forms team. At this point code behind is not compiled (which is what @kaniosm is experiencing!) and you have to clean project and restart IDE. 

XF 4.7 might build fine but it doesn't remove the fact that new functionality on XF 4.7 crashes VS. Is it XF to blame? or VS? Maybe do a call and figure it out. In the meantime I will downgrade to 4.6 and park all the new lovely multi binding in a branch I can merge with at a later point in time.

**Comment by kaniosm** [comment]
@StephaneDelcroix shell routes are fine, it just happened that both routes I was testing were using multibinding, which requires a converter, which was not the case in my proxy implementation.

**Comment by pulmuone** [comment]
xf 4.7 same issue...

**Comment by StephaneDelcroix** [comment]
> shell routes are fine

I'm relieved (/cc @PureWeen)

**Comment by RamiroGuasti** [comment]
The same using Visual Studio 2019 for Windows.
Xamarin.Forms version: 4.7.0.968+385-sha.9062aad78-azdo.3810621

Error: XamlC error XFC0000 : Cannot resolve type "Application".

**Comment by BlueRaja** [comment]
JFYI, NuGet currently lists 4.7.0.968 as the latest "stable" release.  I think this is a mistake with your CI, since this is clearly a dev-build, not something that has been tested + QA'ed.

**Comment by ederjbezerra** [comment]
same issue for m

> xf 4.7 same issue...

For me too:
XamlC error XFC0000 : Cannot resolve type "ContentPage".
Xamarin.Forms 4.7 with VS for mac 8.4

**Comment by Tommigun1980** [comment]
Same here.

**Comment by fmacbeth** [comment]
Same problem with v4.7.0.968 downgraded to v4.6.0.967 and problem goes away. First appeared when I set a resource dictionary in the App.xaml file

**Comment by rscholey** [comment]
Any update on this yet?

**Comment by hyokosdeveloper** [comment]
same here - downgrading back to v.4.6.0.967

**Comment by hyokosdeveloper** [comment]
![FMHH_Utils_–_Visual_Studio_Community_2019_for_Mac](https://user-images.githubusercontent.com/1407528/85380425-ed138000-b50a-11ea-818c-854f3c70d7a1.png)

**Comment by KaraZL** [comment]
Well, I suppose I'll downgrade as well.

**Comment by echolumaque** [comment]
same error as well. when i first open VS, the error exists. but when I restart the project (changes in C# code), the error shows up 
![image](https://user-images.githubusercontent.com/61021721/85761333-3bac5e80-b745-11ea-91af-82c6f8f97c16.png)

**Comment by oliverbraunsdst** [comment]
Maybe this helps a little:
I upgraded my project to 4.7.0.968 a few days ago and didn't have this error until I added a "&lt;" and "&gt;" to one of my XAML files. This triggered the error. It's not disappearing since. Removing the special characters again also didn't help.

**Comment by oliverbraunsdst** [comment]
> Maybe this helps a little:
> I upgraded my project to 4.7.0.968 a few days ago and didn't have this error until I added a "<" and ">" to one of my XAML files. This triggered the error. It's not disappearing since. Removing the special characters again also didn't help.

Sorry, GitHub parsed my post: I added the special syntax for "<" and ">" to my XAML as described here:
https://docs.microsoft.com/de-de/dotnet/framework/wpf/advanced/how-to-use-special-characters-in-xaml

**Comment by ErikSJan** [comment]
I have this same problem since 4.7 but here is the weird thing:
- if I use x86 or x64 configuration I run into this problem and also "XamlC error : La valeur ne peut pas être null." on every xaml file.
- if I use Any CPU configuration, everything compiles fine.

**Comment by rkshnair** [comment]
@BlueRaja @samhouts  This issue is not resolved yet. 
I verified the issue is still present in XF 4.7.0.968. Please reopen the bug.

**Comment by microwales** [comment]
same issues but does not affect compilations

**Comment by rmarinho** [comment]
The fix will be in the next service release we are pushing soon.

**Comment by excellapay** [comment]
I had the same issue. But then deleted FORM and dependent package. Then in GIT, Rolled back the changes to Packages, etc.  That then restored them to Forms, IOS, Droid.  So - My FORMs version is back to the 4.7.0.  Closed SLN, re-opened, Had to say RELOAD on a few projects.  For some reason a few IDs in my SLN updated, but I am back building again.

**Comment by pbiajoni** [comment]
=(

![image](https://user-images.githubusercontent.com/19736827/86070360-e4630280-ba52-11ea-9789-7860d19ec652.png)

**Comment by thanderOliveira** [comment]
I had to Downgrade to Xamarin.Forms 4.6.0.967

**Comment by weeswings** [comment]
Can confirm it's fixed for me with Xamarin.Forms 4.7.0. 1080 
Cheers!

**Comment by jwoeste** [comment]
I can confirm this issue is fixed with Xamarin.Forms 4.7.0.1080 for my project as well.

Great job guys!!!

**Comment by oliverbraunsdst** [comment]
Can confirm it as well: Fixed in 4.7.0.1080. Good work!

**Comment by microwales** [comment]
I can also confirm the fix in version 4.7.0.1080 as well. Thank you guys

**Comment by andersonvieiragomeslopes** [comment]
Can confirm it as well: Fixed in 4.7.0.1080. Good Job!

**Comment by dfyx** [comment]
Still happens for me in 4.7.0.1179 with a new .Net Core 3.1 project.

**Comment by StephaneDelcroix** [comment]
@dfyx please open a new issue, with an attached repro. thx

**Comment by dfyx** [comment]
Just got it, I was missing a line from the project file that removes the EmbeddedResource build action from WPF XAML files. That could be a little clearer from the documentation. I'll add a new issue as soon as I'm sure about the details.

**Comment by lindexi** [comment]
It break in 4.8.0.1269

![](https://user-images.githubusercontent.com/16054566/89698698-30a83900-d955-11ea-8e5d-56fab5ffb8ce.png)

**Comment by lindexi** [comment]
The evil way to solve it: https://github.com/lindexi/lindexi_gd/tree/96c9063fdba9fe318eb099da67422de5cc9ae5af/XamarinNeller/XamarinNeller.WPF

**Comment by hplos** [comment]
Please reopen, this issue returns in 4.8.0.1269

**Comment by Dids** [comment]
Xamarin.Forms 4.8.0.1269 works fine here, but I'm only targeting iOS at the moment (as well as working from macOS + VS4Mac), so maybe the issue is platform specific this time around?

**Comment by kaniosm** [comment]
Xamarin Forms 4.8.0.1269+325-sha.f6490bf7d-azdo.3951545 with VS 2019 on Windows 10 - All fine!

**Comment by TroySchmidt** [comment]
Using .NET Core WPF app the fix above by @schnerring  worked for me as well.

**Comment by Achilles1515** [comment]
Still an issue with 4.8.0.1560 (.NET Core WPF). Workaround mentioned by @schnerring worked.

**Comment by givod** [comment]
it didn't work for me, instead, the whole threw all sorts of errors

**Comment by dcuccia** [comment]
Not sure this is resolved. @schnerring 's work-around worked for me as well on 4.8.0.1687. My scenario is deploying MobileBlazorBindings hybrid components in an existing WPF app. I only observed this issue when adding to my existing project with vanilla WPF UserControls. The source is here:

https://github.com/dcuccia/MobileBlazorBindings/tree/master/samples/HybridAppStandalone

**Comment by Kcops11** [comment]
Currently experiencing this issue as well. Ive dotted my I's and crossed my T's but cant seem to find any obvious reason. Will try to downgrade to fix.

**Comment by cris-m** [comment]
I am experiencing this issue after Visual studio for Mac update to version 8.8.6 (build 15) and Xamarin Forms 5.0.0.1905. Is there any fix without downgrading Xamarin Forms ?

**Comment by Tommigun1980** [comment]
For anyone experiencing this issue please open a new ticket. I am not sure if the closed issues are monitored.

**Comment by schnerring** [comment]
> I am not sure if the closed issues are monitored.

#13217

**Comment by jwoeste** [comment]
> This is still an error and it's two years later so it says a lot about the team developing Visual Studio. I am using VS 2022. Now the error occurred when I made a small C# code change so most of the excuses above are no good. I need to close / open VS 2022 to get rid of this problem. Along with the horrible OS Win 11 MS has been doing a very poor job in the last while.

I had this issue recently as well. Reason was a 3rd party NuGet Package using URL for package reference.

If I recall correctly it was rg.plugins.popup.

Before I had 
`xmlns:pages="http://rotorgames.com" 
`(according to their documentation).

Changing it to
`xmlns:pages="namespace:Rg.Plugins.Popup.Pages;assembly=Rg.Plugins.Popup"` solved the issue. I had one more nuget but I forgot which it was.

Basically don't use URLs - use the old way. 

This XamlC error XFC0000 seems to be some kind of a catch-all error for unhandled errors in the XAML compiler. 

Hope it helps.

**Comment by GreatBarrier86** [comment]
I haven't tried it, but the link mentioned above had this potential workaround.

https://github.com/xamarin/Xamarin.Forms/issues/11101

**Comment by jl-memosoft** [comment]
> > This is still an error and it's two years later so it says a lot about the team developing Visual Studio. I am using VS 2022. Now the error occurred when I made a small C# code change so most of the excuses above are no good. I need to close / open VS 2022 to get rid of this problem. Along with the horrible OS Win 11 MS has been doing a very poor job in the last while.
> 
> I had this issue recently as well. Reason was a 3rd party NuGet Package using URL for package reference.
> 
> If I recall correctly it was rg.plugins.popup.
> 
> Before I had `xmlns:pages="http://rotorgames.com" `(according to their documentation).
> 
> Changing it to `xmlns:pages="namespace:Rg.Plugins.Popup.Pages;assembly=Rg.Plugins.Popup"` solved the issue. I had one more nuget but I forgot which it was.
> 
> Basically don't use URLs - use the old way.
> 
> This XamlC error XFC0000 seems to be some kind of a catch-all error for unhandled errors in the XAML compiler.
> 
> Hope it helps.

+1

**Comment by ajvanlaningham** [comment]
Still an issue I'm experiencing with MAUI:

Severity	Code	Description	Project	File	Line	Suppression State
Error		 : XamlC error XFC0000 : Cannot resolve type "Application"
Error		 : XamlC error XFC0000 : Cannot resolve type "Application".
Error		 : XamlC error XFC0000 : Cannot resolve type "Shell"	
Error		 : XamlC error XFC0000 : Cannot resolve type "Shell"	
Error		 : XamlC error XFC0000 : Cannot resolve type "ResourceDictionary"	
Error		 : XamlC error XFC0000 : Cannot resolve type "ResourceDictionary"	
Error		 : XamlC error XFC0000 : Cannot resolve type "ResourceDictionary"	
Error		 : XamlC error XFC0000 : Cannot resolve type "ResourceDictionary"	


Solution wont build at all. I've cleaned, rebuilt, closed, reopend, recleaned, rebuilt, restarted my computer, repaired visual studio, restored nugget packages, restarted my computer, done all that again and nothing has fixed any of it.

**Comment by Mercally** [comment]
> > This is still an error and it's two years later so it says a lot about the team developing Visual Studio. I am using VS 2022. Now the error occurred when I made a small C# code change so most of the excuses above are no good. I need to close / open VS 2022 to get rid of this problem. Along with the horrible OS Win 11 MS has been doing a very poor job in the last while.
> 
> I had this issue recently as well. Reason was a 3rd party NuGet Package using URL for package reference.
> 
> If I recall correctly it was rg.plugins.popup.
> 
> Before I had `xmlns:pages="http://rotorgames.com" `(according to their documentation).
> 
> Changing it to `xmlns:pages="namespace:Rg.Plugins.Popup.Pages;assembly=Rg.Plugins.Popup"` solved the issue. I had one more nuget but I forgot which it was.
> 
> Basically don't use URLs - use the old way.
> 
> This XamlC error XFC0000 seems to be some kind of a catch-all error for unhandled errors in the XAML compiler.
> 
> Hope it helps.

That work around helped me. Thank you!
